tests/posix/fs/src/test_fat_mount.c

```diff
@@ -18,17 +18,29 @@ static struct fs_mount_t fatfs_mnt = {
 	.fs_data = &fat_fs,
 };
 
-static int test_mount(void)
+void *test_mount(void)
 {
 	int res;
 
 	res = fs_mount(&fatfs_mnt);
 	if (res < 0) {
 		TC_PRINT("Error mounting fs [%d]\n", res);
-		return TC_FAIL;
+		zassert_equal(0, res, NULL);
 	}
 
-	return TC_PASS;
+	return NULL;
+}
+
+void test_unmount(void *unused)
+{
+	int res;
+
+	ARG_UNUSED(unused);
+	res = fs_unmount(&fatfs_mnt);
+	if (res < 0) {
+		TC_PRINT("Error unmounting fs [%d]\n", res);
+		zassert_equal(0, res, NULL);
```

Comment:
Same here, this doesn't run in the test thread, so I'm actually not sure what happens if it fails.

Reply:
I'm just going to use `__ASSERT_NO_MSG()` here until there is some resolution. Is there an open issue number I can reference in comments?